python_ta/checkers/unnecessary_indexing_checker.py

```diff
@@ -34,9 +34,16 @@ def visit_for(self, node: nodes.For) -> None:
             args = node.target.name, iterable
             self.add_message("unnecessary-indexing", node=node.target, args=args)
 
+    @only_required_for_messages("unnecessary-indexing")
+    def visit_comprehension(self, node: nodes.Comprehension) -> None:
+        iterable = _iterable_if_range(node.iter)
+        if iterable is not None and _is_unnecessary_indexing(node):
+            args = node.target.name, iterable
+            self.add_message("unnecessary-indexing", node=node.target, args=args)
+
 
 # Helper functions
-def _is_unnecessary_indexing(node: nodes.For) -> bool:
+def _is_unnecessary_indexing(node: Union[nodes.For, nodes.Comprehension]) -> bool:
     """Return whether the iteration variable in the for loop is ONLY used to index the iterable.
```

Comment:
This docstring needs to be updated as well.

In general, we need to make sure to distinguish between "for loop" and "comprehension", since a comprehension isn't a loop. So throughout, where the existing documentation says "for loop" you should replace it with "for loop/comprehension" instead.